clients/slo/slo_test.go

```diff
@@ -0,0 +1,465 @@
+// @license
+// Copyright 2025 Dynatrace LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slo_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/rest"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/clients/slo"
+)
+
+func TestNewClient(t *testing.T) {
+	actual := slo.NewClient(&rest.Client{})
+	require.IsType(t, slo.Client{}, *actual)
+}
+
+func TestList(t *testing.T) {
+	t.Run("successfully returned all configuration from server", func(t *testing.T) {
+		apiResponse1 := `{
+  "totalCount": 2,
+  "nextPageKey": "key_for_next_page",
+  "slos": [
+    {
+      "name": "CPU utilization",
+      "description": "Measures the CPU usage of selected hosts over time.",
+      "sliReference": "reference to slo template",
+      "criteria": [
+        {
+          "timeframeFrom": "now-7d",
+          "timeframeTo": "now",
+          "target": 98
+        }
+      ],
+      "tags": [],
+      "id": "slo-id-1",
+      "version": "ver1"
+    }
+  ]
+}`
+		apiResponse2 := `{
+  "totalCount": 2,
+  "slos": [
+    {
+      "name": "K8s namespace memory requests efficiency - OLD",
+      "description": "Compares the actual usage of memory to the requested memory.",
+      "customSli": "DQL based SLO",
+      "criteria": [
+        {
+          "timeframeFrom": "now-7d",
+          "timeframeTo": "now",
+          "target": 80,
+          "warning": 90
+        }
+      ],
+      "tags": [],
+      "id": "slo-id-2",
+      "version": "ver1"
+    }
+  ]
+}`
+
+		server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			require.Equal(t, http.MethodGet, request.Method)
```

Comment:
nit: the boilerplate code of the test is the same everywhere, why not inject the behavior in and prevent the repetition?